src/framework/utils/entity-reference.js

```diff
@@ -266,10 +266,12 @@ class EntityReference extends EventHandler {
             nextEntityGuid = nextEntity.getGuid();
             this._parentComponent.data[this._entityPropertyName] = nextEntityGuid;
         } else {
-            const root = this._parentComponent.system.app.root;
-            const isOnSceneGraph = this._parentComponent.entity.isDescendantOf(root);
+            // If value is the GUID, look for the entity from the root of the parent entity.
+            // We use entity.root rather than app.root because the parent entity may be detached
+            // from the scene, and the value is referencing an entity within this detached graph.
+            const root = this._parentComponent.entity.root;
 
-            nextEntity = (isOnSceneGraph && nextEntityGuid) ? root.findByGuid(nextEntityGuid) : null;
+            nextEntity = nextEntityGuid ? root.findByGuid(nextEntityGuid) : null;
```

Comment:
much cleaner. How about something like this, to search in both scene and entity trees, to increase likelihood of finding it?

```
let nextEntity = null;
if (nextEntityGuid instanceof Entity) {
    // if value is set to a Entity itself replace value with the GUID
    nextEntity = nextEntityGuid;
    nextEntityGuid = nextEntity.getGuid();
    this._parentComponent.data[this._entityPropertyName] = nextEntityGuid;
} else if (nextEntityGuid) {
    const sceneRoot = this._parentComponent.system.app.root;
    const entityRoot = this._parentComponent.entity.root;

    nextEntity = entityRoot.findByGuid(nextEntityGuid) || sceneRoot.findByGuid(nextEntityGuid);
}
```